app.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/reloader'
+require 'json'
+require 'cgi'
+
+FILE_PATH = 'public/memos.json'
+
+def get_memos(file_path)
+  File.open(file_path) { |f| JSON.parse(f.read) }
+end
+
+def set_memos(file_path, memos)
+  File.open(file_path, 'w') { |f| JSON.dump(memos, f) }
+end
+
+get '/' do
+  redirect '/memos'
+end
+
+get '/memos' do
+  @app_name = 'メモアプリ'
+  @memos = get_memos(FILE_PATH)
+  erb :index
+end
+
+get '/memos/new' do
+  erb :new
+end
+
+get '/memos/:id/edit' do
+  memos = get_memos(FILE_PATH)
+  @title = memos[params[:id]]['title']
+  @content = memos[params[:id]]['content']
```

Comment:
```suggestion
  @memo = memos[params[:id]]
```

項目が増えるとインスタンス変数がどんどん増えていくので、memoだけをerbに渡しましょう

あと、

```ruby
memos = get_memos(FILE_PATH)
```

で常に全件返すのではなく、

```ruby
@memo = hogehoge(params[:id])
```

みたいな形で、1件だけmemoを返すメソッドがあった方が便利そうです